numpy/core/_methods.py

```diff
@@ -140,13 +151,13 @@ def _clip(a, min=None, max=None, out=None, *, casting=None, **kwargs):
         return _clip_dep_invoke_with_casting(
             um.clip, a, min, max, out=out, casting=casting, **kwargs)
 
-def _mean(a, axis=None, dtype=None, out=None, keepdims=False):
+def _mean(a, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
     arr = asanyarray(a)
 
     is_float16_result = False
-    rcount = _count_reduce_items(arr, axis)
-    # Make this warning show up first
-    if rcount == 0:
+
+    rcount = _count_reduce_items(arr, axis, keepdims=keepdims, where=where)
+    if umr_any(rcount == 0, axis=None):
```

Comment:
Isn't none the default?

Reply:
Nevermind, see you mention that above.

Reply:
This is what I thought, because for `np.any`, the default is `axis=None`. However when the checks failed, I investigated it and found that for `numpy.ufunc.reduce`, the default is `axis=0`
[Version 1.19.0 `numpy.ufunc.reduce` doc](https://numpy.org/devdocs/reference/generated/numpy.ufunc.reduce.html)

Reply:
I don't know if this difference is intentional. Maybe it would make sense to harmonize this in another issue in the future to increase the consistency.